services/web/server/src/simcore_service_webserver/session/plugin.py

```diff
@@ -15,6 +16,45 @@
 _logger = logging.getLogger(__name__)
 
 
+class SharedCookieEncryptedCookieStorage(EncryptedCookieStorage):
+    async def save_session(
+        self,
+        request: web.Request,
+        response: web.StreamResponse,
+        session: aiohttp_session.Session,
+    ) -> None:
+        # link response to originating request (allows to detect the orginal request url)
+        response._req = request  # pylint:disable=protected-access  # noqa: SLF001
+
+        await super().save_session(request, response, session)
+
+    def save_cookie(
+        self,
+        response: web.StreamResponse,
+        cookie_data: str,
+        *,
+        max_age: int | None = None,
+    ) -> None:
+        params = self._cookie_params.copy()
+
+        # share cookie accross all subdomains
+        # overwrite domain from `None` (browser sets `example.com`) to `.example.com`
+        request = response._req  # pylint:disable=protected-access  # noqa: SLF001
+        assert isinstance(request, web.Request)  # nosec
+        params["domain"] = f".{request.url.host}"
+
+        if max_age is not None:
+            params["max_age"] = max_age
+            t = time.gmtime(time.time() + max_age)
```

Comment:
Your suggested approach might have some unexpected side effects since you are altering the global object which this function specifically makes a copy of.

Reply:
I've added a warning to be on the save side